pkg/services/searchV2/index.go

```diff
@@ -846,30 +846,96 @@ func (l sqlDashboardLoader) loadAllDashboards(ctx context.Context, limit int, or
 			dashboardQuerySpan.SetAttributes("dashboardUID", dashboardUID, attribute.Key("dashboardUID").String(dashboardUID))
 			dashboardQuerySpan.SetAttributes("lastID", lastID, attribute.Key("lastID").Int64(lastID))
 
-			rows := make([]*dashboardQueryResult, 0)
+			var slices [][]string
 			err := l.sql.WithDbSession(dashboardQueryCtx, func(sess *db.Session) error {
-				sess.Table("dashboard").
-					Where("org_id = ?", orgID)
+				sql := "select id, uid, is_folder, folder_id, slug, data, created, updated from dashboard where org_id = ?"
+				sqlAndArgs := []interface{}{"", orgID}
 
 				if lastID > 0 {
-					sess.Where("id > ?", lastID)
+					sql += " AND id > ?"
+					sqlAndArgs = append(sqlAndArgs, lastID)
 				}
 
 				if dashboardUID != "" {
-					sess.Where("uid = ?", dashboardUID)
+					sql += " AND uid = ?"
+					sqlAndArgs = append(sqlAndArgs, dashboardUID)
 				}
 
-				sess.Cols("id", "uid", "is_folder", "folder_id", "data", "slug", "created", "updated")
+				sql += " order by id asc"
+				sql += " limit ?"
+				sqlAndArgs = append(sqlAndArgs, limit)
 
-				sess.OrderBy("id ASC")
-				sess.Limit(limit)
-
-				return sess.Find(&rows)
+				sqlAndArgs[0] = sql
+				output, err := sess.QuerySliceString(sqlAndArgs...)
+				slices = output
+				return err
 			})
 
+			dashboardQuerySpan.SetAttributes("dashboardCount", len(slices), attribute.Key("dashboardCount").Int(len(slices)))
+
+			if err != nil || slices == nil {
+				dashboardQuerySpan.End()
+				ch <- &dashboardsRes{
+					dashboards: nil,
+					err:        err,
+				}
+				break
+			}
+
+			rows := make([]*dashboardQueryResult, len(slices))
+			var parsingErr error
+			for i := range slices {
+				id, err := strconv.ParseInt(slices[i][0], 10, 64)
+				if err != nil {
+					parsingErr = err
+					break
+				}
+				uid := slices[i][1]
+				isFolder := false
+				if slices[i][2] == "1" {
```

Comment:
not sure if this is checked elsewhere, but it might be worth checking the length of the slices to avoid panics

Reply:
added https://github.com/grafana/grafana/pull/57509/commits/b1679b4067eb237b847c35f237c93fde0bef0d66